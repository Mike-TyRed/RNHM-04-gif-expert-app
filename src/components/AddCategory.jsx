import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {// onNewCategory es una función que se recibe como argumento
  //Hook de useState para manejar el estado del input
  const [inputValue, setInputValue] = useState("");//El valor inicial del input es un string vacio

  //Función para manejar el evento onChange del input
  const handleInputChange = (e) => {//handleInputChange es una función que se recibe como argumento
    setInputValue(e.target.value);//El valor del input es el valor del evento
  };

  const onSubmit = (e) => {//onSubmit es una función que se recibe como argumento
    //console.log("Submit hecho");
    e.preventDefault();//Previene el comportamiento por defecto del formulario

    if (inputValue.trim().length > 1) {//Si el valor del input es mayor a 1
      //setCategories((cats) => [...cats, inputValue]);
      onNewCategory(inputValue.trim());//Se ejecuta la función onNewCategory que se recibe como argumento
      setInputValue("");//El valor del input es un string vacio
    }
  };

  return (
    <form onSubmit={(e) => onSubmit(e)} aria-label="form"> 
    <input
        type="text"
        placeholder="Buscar gifs..."
        value={inputValue}
        onChange={(e) => handleInputChange(e)}
      ></input>
    </form>//Se ejecuta la función onSubmit que se recibe como argumento
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,//onNewCategory es una función que se recibe como argumento
};