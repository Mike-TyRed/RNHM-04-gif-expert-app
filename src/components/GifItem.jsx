import PropTypes from "prop-types";

export const GifItem = ({title, url, id}) => {
    
  return (
  <div className="card">
    <p>{title}</p>
    <img src={url} alt={title} />
  </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired, // El titulo es requerido y debe ser un string
  url: PropTypes.string.isRequired, // La url es requerida y debe ser un string
};