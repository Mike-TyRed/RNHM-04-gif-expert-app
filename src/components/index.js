export * from './GifGrid';
export * from './GifItem';
export * from './AddCategory';