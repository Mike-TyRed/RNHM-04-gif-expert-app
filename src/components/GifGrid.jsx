import { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  /* const [images, setImages] = useState([]);
  const getImages = async () => {
    const newImages = await getGif(category);
    setImages(newImages);
  };

  useEffect(() => {
    getImages(category);
  }, []); */

  const { images, isLoading } = useFetchGifs(category); //<-- Custom Hook

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Loading...</h2>}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
