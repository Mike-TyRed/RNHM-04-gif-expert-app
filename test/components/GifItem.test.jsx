import { render, screen } from "@testing-library/react";
import { exact } from "prop-types";
import { GifItem } from "../../src/components/GifItem"; // Importa el componente a probar

describe("Pruebas en GiftItem", () => {
  const title = "Akira"; // Crea un titulo
  const url = "https://akira.com/akira.jpg"; // Crea una url

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />); // Renderiza el componente a probar
    expect(container).toMatchSnapshot(); // Compara el snapshot con el renderizado
  });

  test("Debe mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />); // Renderiza el componente a probar

    //screen.debug(); // Muestra el componente renderizado en consola

    //expect(screen.getByRole("img").src).toBe(url) // Comprueba que el elemento img tenga el atributo src con el valor de la url
    //expect(screen.getByRole("img").alt).toBe(title) // Comprueba que el elemento img tenga el atributo alt con el valor del titulo

    const { src, alt } = screen.getByRole("img"); // Extrae el src y el alt del elemento img
    expect(src).toBe(url); // Comprueba que el src sea igual a la url
    expect(alt).toBe(title); // Comprueba que el alt sea igual al titulo
  });

  test("Debe mostrar el titulo", () => {
    render(<GifItem title={title} url={url} />); // Renderiza el componente a probar
    expect(screen.getByText(title)).toBeTruthy(); // Comprueba que el titulo este en el documento
  });
});
