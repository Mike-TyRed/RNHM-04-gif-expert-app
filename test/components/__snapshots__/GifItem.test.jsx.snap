// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Akira
    </p>
    <img
      alt="Akira"
      src="https://akira.com/akira.jpg"
    />
  </div>
</div>
`;
